cmd/root.go

```diff
@@ -75,20 +75,86 @@ func initConfig() {
 	nc.Close()
 }
 
+func migrateConfigFiles() bool {
+
+	changed := false
+	//config
+	configFile := config.GetNetclientPath() + "netclient.yml"
+	if _, err := os.Stat(configFile); err == nil {
+		netclientl := config.Config{}
+		f, err := os.Open(configFile)
+		if err != nil {
+			slog.Error("error opening the config.yml file", "error", err.Error())
+		}
+		if err = yaml.NewDecoder(f).Decode(&netclientl); err != nil {
+			slog.Error("error decoding the config.yml file", "error", err.Error())
+		}
```

Comment:
if decoding fails shouldn't proceed further for this config file

Reply:
Change done.  If open/decode failed, it will return error. 
Also added more error check, if one of config file(config,node or server) migration failed, the migration will return error, and it will block the startup.  Migration check criteria is,  without migration or all migration((config,node or server)) successfully.
Please help review the code changes,  also the migration check logic.